In [1]:
import numpy as np
import math
import matplotlib.pyplot as plt
import scipy.stats as stats
from scipy.stats import probplot
import statsmodels.api as sm

# 6.5
H_o:$C*\mu=0$ vs $H_a:C*\mu$ is not equal to 0.

In [28]:
#a)
n=40
q=3
x_bar=np.array([[46.1],[57.3],[50.4]])
Covmat=np.array([[101.3,63,71],[63,80.2,55.6],[71,55.6,97.4]])
C=np.array([[1,-1,0],[1,0,-1]])
T_squared=n*np.matmul((np.matmul((np.matmul(C,x_bar)).T,np.linalg.inv(np.dot(np.dot(C,Covmat),C.T)))),np.matmul(C,x_bar))
print("Test Statistic:",float(T_squared[0]))

#find F critical value
crit_val=(stats.f.ppf(q=1-.05, dfn=2, dfd=38))*(((n-1)*(q-1))/(n-q+1))
print("Critical Value:",crit_val)

print("Since Test Statistic > Critical Value, we know that we must reject the null and conlcude that mean indices are unequal.")

Test Statistic: 90.49457756898148
Critical Value: 6.660416635188395
Since Test Statistic > Critical Value, we know that we must reject the null and conlcude that mean indices are unequal.


In [65]:
#b)
simul_conflow1=np.matmul(C[0],x_bar)-(math.sqrt(crit_val)*(math.sqrt(np.matmul((np.matmul(C[0].T,Covmat)),C[0])/n)))
simul_confhigh1=np.matmul(C[0],x_bar)+(math.sqrt(crit_val)*(math.sqrt(np.matmul((np.matmul(C[0].T,Covmat)),C[0])/n)))
print("Confidence Interval for mu1-mu2:(",float(simul_conflow1),",",float(simul_confhigh1),")")

simul_conflow2=np.matmul(C[1],x_bar)-(math.sqrt(crit_val)*(math.sqrt(np.matmul((np.matmul(C[1].T,Covmat)),C[1])/n)))
simul_confhigh2=np.matmul(C[1],x_bar)+(math.sqrt(crit_val)*(math.sqrt(np.matmul((np.matmul(C[1].T,Covmat)),C[1])/n)))
print("Confidence Interval for mu1-mu3:(",float(simul_conflow2),",",float(simul_confhigh2),")")

Confidence Interval for mu1-mu2:( -14.239955276204547 , -8.160044723795444 )
Confidence Interval for mu1-mu3:( -7.372643907187999 , -1.2273560928119958 )
Since 0 does not belong in either of these intervals, we conclude that mean indices are not equal


Since 0 does not belong in either of these intervals, our conclusion from above is reinforced.

# 6.6

In [2]:
#a)
n1=3
n2=4
Tr2_1=np.array([[3],[3]])
Tr2_2=np.array([[1],[6]])
Tr2_3=np.array([[2],[3]])
Tr3_1=np.array([[2],[3]])
Tr3_2=np.array([[5],[1]])
Tr3_3=np.array([[3],[1]])
Tr3_4=np.array([[2],[3]])
x_bar_1=(1/n1)*(np.add(np.add(Tr2_1,Tr2_2),Tr2_3))
x_bar_2=(1/n2)*(np.add(np.add(np.add(Tr3_1,Tr3_2),Tr3_3),Tr3_4))
s_1=(.5)*np.add(np.add((np.matmul(np.subtract(Tr2_1,x_bar_1),np.subtract(Tr2_1,x_bar_1).T)),
                         (np.matmul(np.subtract(Tr2_2,x_bar_1),np.subtract(Tr2_2,x_bar_1).T))),
                         (np.matmul(np.subtract(Tr2_3,x_bar_1),np.subtract(Tr2_3,x_bar_1).T)))

s_2=(1/(n2-1))*np.add(np.add(np.add((np.matmul(np.subtract(Tr3_1,x_bar_2),np.subtract(Tr3_1,x_bar_2).T)),
                         (np.matmul(np.subtract(Tr3_2,x_bar_2),np.subtract(Tr3_2,x_bar_2).T))),
                         (np.matmul(np.subtract(Tr3_3,x_bar_2),np.subtract(Tr3_3,x_bar_2).T))),
                     (np.matmul(np.subtract(Tr3_4,x_bar_2),np.subtract(Tr3_4,x_bar_2).T)))
S_pooled=np.add(((n1-1)/(n1+n2-2))*s_1,((n2-1)/(n1+n2-2))*s_2)
print(S_pooled)

[[ 1.6 -1.4]
 [-1.4  2. ]]


$H_o:\mu_{2}-\mu_{3}=0$ vs $H_a:\mu_{2}-\mu_{3}$ is not equal to 0.

In [114]:
#b
T_squared_6=np.matmul(np.matmul(np.subtract(x_bar_1,x_bar_2).T,np.linalg.inv(
((1/3)+(1/4))*S_pooled)),np.subtract(x_bar_1,x_bar_2))
print("Test Statistic:",float(T_squared_6))
crit_val_6=(stats.f.ppf(q=1-.01, dfn=2, dfd=4))*(10/4)
print("Critical Value:",crit_val_6)
print("Since our test statistic is much smaller than the critical value we cannot reject the null hypothesis and must conclude that mu2 and mu3 are equal.")

Test Statistic: 3.870967741935484
Critical Value: 44.99999999999997
Since our test statistic is much smaller than the critical value we cannot reject the null hypothesis and must conclude that mu2 and mu3 are equal.


In [134]:
#c
simul_conflow6_1=np.subtract(x_bar_1[0],x_bar_2[0])-(math.sqrt(crit_val_6))*math.sqrt((((1/3)+(1/4))*S_pooled[0,0]))
simul_confhigh6_1=np.subtract(x_bar_1[0],x_bar_2[0])+(math.sqrt(crit_val_6))*math.sqrt((((1/3)+(1/4))*S_pooled[0,0]))
print("Confidence Interval mu_21 - mu_31:(",float(simul_conflow6_1),",",float(simul_confhigh6_1),")")
simul_conflow6_2=np.subtract(x_bar_1[1],x_bar_2[1])-(math.sqrt(crit_val_6))*math.sqrt((((1/3)+(1/4))*S_pooled[1,1]))
simul_confhigh6_2=np.subtract(x_bar_1[1],x_bar_2[1])+(math.sqrt(crit_val_6))*math.sqrt((((1/3)+(1/4))*S_pooled[1,1]))
print("Confidence Interval mu_22 - mu_32:(",float(simul_conflow6_2),",",float(simul_confhigh6_2),")")
print("This result reinforces the hypothesis test from part b) since 0 is contained in both of these intervals")

Confidence Interval mu_21 - mu_31:( -7.480740698407858 , 5.480740698407858 )
Confidence Interval mu_22 - mu_32:( -5.245688373094716 , 9.245688373094715 )
This result reinforces the hypothesis test from part b) since 0 is contained in both of these intervals


# 6.7

$H_o:\mu_{1}-\mu_{2}=0$ vs $H_a:\mu_{1}-\mu_{2}$ is not equal to 0.

In [140]:
x_bar_1_7=np.array([[204.4],[556.6]])
x_bar_2_7=np.array([[130],[355]])
S_1_7=np.array([[13825.3,23823.4],[23823.4,73107.4]])
S_2_7=np.array([[8632,19616.7],[19616.7,55964.5]])
S_pooled_7=np.add((44/98)*S_1_7,(54/98)*S_2_7)
T_squared_7=np.matmul(np.matmul(np.subtract(x_bar_1_7,x_bar_2_7).T,np.linalg.inv(
((1/45)+(1/55))*S_pooled_7)),np.subtract(x_bar_1_7,x_bar_2_7))
print("Test Statistic:",float(T_squared_7))
crit_val_7=(stats.f.ppf(q=1-.05, dfn=2, dfd=97))*(196/97)
print("Critical Value:",crit_val_7)
print("Since the Test Statistic is greater than the critical value, we have enough evidence to reject the null hypothesis. Thus, mu1 and mu2 are not equal.")

Test Statistic: 16.066219085641638
Critical Value: 6.244088539488168
Since the Test Statistic is greater than the critical value, we have enough evidence to reject the null hypothesis. Thus, mu1 and mu2 are not equal.


In [141]:
#coefficient vector
print(np.matmul(np.linalg.inv(S_pooled_7),np.subtract(x_bar_1_7,x_bar_2_7)))

[[0.00170252]
 [0.00259163]]


By remark on page 289, we know that the linear combination $\hat{a}^{'}\left ( \bar{x_{1}} - \bar{x_{2}} \right )$ with coefficient vector $ \bigl(\begin{smallmatrix}
 0.00170252 \\ 0.00259163\end{smallmatrix}\bigr)$ quantifies the largest population difference.

# 6.8

In [3]:
#c
crit_val_8=(stats.f.ppf(q=1-.01, dfn=4, dfd=16))
print(crit_val_8)

4.772577999723211


In [7]:
crit_val_8Barlett=(stats.chi2.ppf(.99, 4))
print(crit_val_8Barlett)

13.276704135987622


# 6.26

In [25]:
x_bar_1_26=np.array([[.153],[-.231],[-.322],[-.339]])
x_bar_2_26=np.array([[.151],[.18],[.256],[.257]])
S_pooled_26=np.array([[.804,.355,.228,.232],[.355,.722,.233,.199],[.228,.233,.592,.239],[.232,.199,.239,.479]])
C_26=np.array([[-1,1,0,0],[0,-1,1,0],[0,0,-1,1]])
T_squared_xC=np.matmul(np.subtract(x_bar_1_26,x_bar_2_26).T,C_26.T)
T_squared_CSC=np.linalg.inv(((1/28)+(1/58))*np.matmul(np.matmul(C_26,S_pooled_26),C_26.T))
T_squared_26=np.matmul(np.matmul(T_squared_xC,T_squared_CSC),T_squared_xC.T)
crit_val_26=3.07317*((stats.f.ppf(q=1-.05, dfn=3, dfd=52)))
print("Test Statistic:",float(T_squared_26))
print("Critical Value:",float(crit_val_26))

Test Statistic: 9.235720861669341
Critical Value: 8.55140414328653


Since our Test statistic is greater than the critical value, we must reject the null hypothesis. So we conclude that there is not parallelism in the profiles.